modules/xmpp/JingleSessionPC.js

```diff
@@ -2251,18 +2205,18 @@ export default class JingleSessionPC extends JingleSession {
      * Closes the peerconnection.
      */
     close() {
-        this.closed = true;
+        this.state = JingleSessionState.ENDED;
+        this.establishmentDuration = undefined;
 
-        // The signaling layer will remove it's listeners
-        this.signalingLayer.setChatRoom(null);
+        this.modificationQueue.push(finishCallback => {
```

Comment:
Have you considered removing all tasks from queue before adding the close there and shutting it down? Do you think something would break if we do it?

Reply:
the queue has a method `kill` which does this.  it also clears the `drain` handler, but we don't currently rely on that.  i wish it had a better name, though.  it feels a little weird to call `kill` and then add another task.

Reply:
I will try to see if I can kill the queue and then have the end task to execute

Reply:
yes, it worked